modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj

```diff
@@ -78,29 +78,31 @@
    ;; we don't care how many pages it took to load this dataset above. it will be a large
    ;; number because we're just tracking the number of times `get-query-results` gets invoked.
 
-   ;; TODO Temporarily disabling due to flakiness (#33140)
-   #_
    (testing "with pagination"
      (let [pages-retrieved (atom 0)
            page-callback   (fn [] (swap! pages-retrieved inc))]
-       (with-bindings {#'bigquery/*page-size*             25
-                       #'bigquery/*page-callback*         page-callback}
-         (let [actual (->> (metadata-queries/table-rows-sample (t2/select-one Table :id (mt/id :venues))
-                             [(t2/select-one Field :id (mt/id :venues :id))
-                              (t2/select-one Field :id (mt/id :venues :name))]
-                             (constantly conj))
-                           (sort-by first)
-                           (take 5))]
+       (with-bindings {#'bigquery/*page-size*     25
+                       #'bigquery/*page-callback* page-callback}
+         (let [results (->> (metadata-queries/table-rows-sample (t2/select-one Table :id (mt/id :venues))
+                              [(t2/select-one Field :id (mt/id :venues :id))
+                               (t2/select-one Field :id (mt/id :venues :name))]
+                              (constantly conj))
+                            (sort-by first)
+                            (take 5))]
            (is (= [[1 "Red Medicine"]
                    [2 "Stout Burgers & Beers"]
                    [3 "The Apple Pan"]
                    [4 "Wurstküche"]
                    [5 "Brite Spot Family Restaurant"]]
-                  actual))
+                  (take 5 results)))
+           (testing "results are not duplicated when pagination occurs (#45953)"
+             (is (= (count results) (count (distinct results)))))
            ;; the `(sort-by)` above will cause the entire resultset to be realized, so
            ;; we want to make sure that it really did retrieve 25 rows per request
            ;; this only works if the timeout has been temporarily set to 0 (see above)
-           (is (= 4 @pages-retrieved))))))))
+
+           ;; TODO Temporarily disabling due to flakiness (#33140)
```

Comment:
This was the part that was still flaking... for the time being I just moved the comment here.